File: src/CSharp/326.IsPowerOfThree.cs
using System;
using System.Linq;
using System.Collections.Generic;

namespace PeefyLeetCode.IsPowerOfThree
{
public class Solution {
    public bool IsPowerOfThree(int n) {
        if (n < 1)
            return false;
        while (n % 3 == 0){ 
            n /= 3;
        }
        return n == 1;
    }
}
}

File: src/Python/326.IsPowerOfThree.py

class Solution:
    def isPowerOfThree(self, n):
        """
        :type n: int
        :rtype: bool
        """
        return n > 0 and 1162261467 % n == 0

if __name__ == '__main__':
    solution = Solution()
    print(solution.isPowerOfThree(3))
    print(solution.isPowerOfThree(9))
    print(solution.isPowerOfThree(0))
    print(solution.isPowerOfThree(45))
else:
    pass
